ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -115,16 +98,160 @@ object ParallelIndexerFactory {
             ).map(_ => ())
 
       def subscribe(readService: ReadService): Future[Source[Unit, NotUsed]] =
-        postgresDaoPool
-          .execute(_.initialize)
-          .map { case (offset, eventSeqId) =>
-            ingest(eventSeqId.getOrElse(0L))(readService.stateUpdates(beginAfter = offset))
-          }(scala.concurrent.ExecutionContext.global)
+        dbDispatcher
+          .executeSql(metrics.daml.parallelIndexer.initialization)(
+            storageBackend.initialize
+          ) // TODO here is an implicit LoggingContext injected from the factory apply implicit param, is that good this way? shouldn't we create here something for the purpose?
+          .map(initialized =>
+            ingest(initialized.lastEventSeqId.getOrElse(0L))(
+              readService.stateUpdates(beginAfter = initialized.lastOffset)
+            )
+          )(scala.concurrent.ExecutionContext.global)
 
       toIndexer(subscribe)(mat)
     }
   }
 
+  case class RunningBatch[T](
```

Comment:
It's more a "opaque batch with some metadata about the indexing pipeline" than `RunningBatch`, but I would not worry too much about the naming.

Reply:
@mziolekda suggested simply batch...I am ok with that as well until we find something better. So I am planning to refactor towards Batch for now, but as soons as you would have an enlightening moment of a good name for this please come forward :)